pyfr/solvers/baseadvec/system.py

```diff
@@ -61,3 +84,45 @@ def deps(dk, *names): return self._kdeps(k, dk, *names)
         g2.commit()
 
         return g1, g2
+
+    @memoize
+    def _preproc_graphs(self, uinbank):
+        m = self._mpireqs
+        k, _ = self._get_kernels(uinbank, None)
+
+        def deps(dk, *names): return self._kdeps(k, dk, *names)
+
+        g1 = self.backend.graph()
+
+        # Interpolate the solution to the flux points
+        if 'eles/local_entropy' in k:
+            g1.add_all(k['eles/disu'])
+
+        # Compute local minimum entropy within element
+        g1.add_all(k['eles/local_entropy'])
+
+        # Pack and send the entropy values to neighbors
+        g1.add_all(k['mpiint/ent_fpts_pack'], deps=k['eles/local_entropy'])
+        for send, pack in zip(m['ent_fpts_send'], k['mpiint/ent_fpts_pack']):
+            g1.add_mpi_req(send, deps=[pack])
+
+        # Compute common entropy minima at internal/boundary interfaces
+        g1.add_all(k['iint/comm_entropy'], deps=k['eles/local_entropy'])
+        g1.add_all(k['bcint/comm_entropy'],
+                   deps=k['eles/local_entropy'] + k['eles/disu'])
+        g1.commit()
+
+        g2 = self.backend.graph()
+
+        # Compute common entropy minima at MPI interfaces
+        g2.add_all(k['mpiint/ent_fpts_unpack'])
+        for l in k['mpiint/comm_entropy']:
+            g2.add(l, deps=deps(l, 'mpiint/ent_fpts_unpack'))
+        g2.commit()
+
+        return g1, g2
```

Comment:
Only return graphs if they are populated with kernels.  Otherwise an empty tuple will do (and saves some overhead down the line).